src/openms/source/METADATA/AbsoluteQuantitationStandards.cpp

```diff
@@ -28,34 +28,80 @@
 // ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 //
 // --------------------------------------------------------------------------
-// $Maintainer: Douglas McCloskey $
-// $Authors: Douglas McCloskey $
+// $Maintainer: Douglas McCloskey, Pasquale Domenico Colaianni $
+// $Authors: Douglas McCloskey, Pasquale Domenico Colaianni $
 // --------------------------------------------------------------------------
 
 #include <OpenMS/METADATA/AbsoluteQuantitationStandards.h>
 
-#include <cstddef> // for size_t & ptrdiff_t
-#include <vector>
-#include <string>
-
 namespace OpenMS
 {
-    
-    AbsoluteQuantitationStandards::AbsoluteQuantitationStandards()
-    {
-    }
-    
-    AbsoluteQuantitationStandards::~AbsoluteQuantitationStandards()
+  void AbsoluteQuantitationStandards::mapComponentsToConcentrations(
+    const std::vector<AbsoluteQuantitationStandards::runConcentration>& run_concentrations,
+    const std::vector<FeatureMap>& feature_maps,
+    std::map<String, AbsoluteQuantitationStandards::featureConcentration>& components_to_concentrations
+  ) const
+  {
+    components_to_concentrations.clear();
+    for (const FeatureMap& fmap : feature_maps)
     {
+      if (!fmap.metaValueExists("sample_name"))
+      {
+        throw Exception::MissingInformation(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "The FeatureMap misses the 'sample_name' MetaValue information.");
+      }
+      const String sample_name = fmap.getMetaValue("sample_name");
+      for (const Feature& feature : fmap)
+      {
+        const std::vector<Feature>& subordinates = feature.getSubordinates();
+        if (subordinates.size() != 2)
```

Comment:
The size check of `subordinates` can be removed.  Some features may have just a single subordinate.